### PR TITLE
[WIP] Expose HTTPS and vhost options from the vue.js web server

### PR DESCRIPTION
**Pull request by onitake**
This PR is a proposed solution for #37 .

Not everybody may want to configure HTTPS, it's probably necessary to make some of the env vars optional.

### CODE REVIEW

File: .env.local.example [hunk 1]
```diff
@@ -1 +1,7 @@
-API_URL=http://localhost:8080/client/api
+API_URL=http://localhost:8080/client/api
```
Comment (rhtyd) [inline .env.local.example]: @onitake Can you rename (or create a new file) this file to .env.local.https.example?

File: vue.config.js [hunk 2]
```diff
@@ -97,7 +98,15 @@ module.exports = {
         ws: false,
         changeOrigin: true
       }
-    }
+    },
+    public: process.env.PUBLIC_HOST,
+    https: {
```
Comment (rhtyd) [inline vue.config.js]: Will this fail in case https certs are not define? For example for normal http:// mgmt server url?

**Comment by onitake** [comment]
I'm pretty sure that startup will fail if the certificates can't be found.
As I don't know JS and Vue very well, some help with making the env vars optional in a clean way is highly appreciated.

**Comment by rhtyd** [comment]
@onitake Please port/push towards new repo https://github.com/apache/cloudstack-primate

**Comment by onitake** [comment]
Will do.

**Pull request closed without merging**